docs/configuration.rst

```diff
@@ -472,3 +477,69 @@ Optional Settings
+
+Required Settings
+"""""""""""""""""
+The following settings must be passed to run the KSQL Server image.
+
+``KSQL_BOOTSTRAP_SERVERS``
+
+To configure the Kafka cluster that KSQL should read from/write to, you must specify one or more "bootstrap" brokers from that Kafka cluster. This parameter is formatted as a comma-separated list of hostname:port pairs; for example, broker-hostname1:9092,broker-hostname2:9092. These brokers are used by KSQL to establish the initial connection to the Kafka cluster and to discover the set of all available brokers in the Kafka cluster, the set of which may change dynamically at runtime (e.g. brokers may be taken down for maintenance). You can but don't need to specify all brokers in the Kafka cluster, though it is recommended to specify at least two brokers to make bootstrapping more resilient against individual broker outages.
+
+Optional Settings
+"""""""""""""""""
+All other settings for KSQL like :ref:`security <ksql-security>`, monitoring interceptors, overrides for Kakfa Streams, Kafka Producer, or Kakfa Consumer settings can be passed to the Docker images as environment variables. The names of these environment variables are derived by prefixing with KSQL_, uppercasing each word in the parameter name, and separating each word with an underscore (_) instead of periods (.). For example, if you need to set ``ksql.streams.state.dir``, the environment variable name would be ``KSQL_KSQL_STREAMS_STATE_DIR``.
+
+The image will then convert these environment variables to corresponding KSQL Server config variables.
+
+--------
+KSQL CLI
+--------
+The KSQL CLI image will execute the ``ksql`` command and try to connect to the server at ``http://localhost:8088`` by default, as shown below. 
+
+.. sourcecode:: bash
+
+  docker run -it -rm confluentinc/cp-ksql-cli:4.1.2
+
+Note: Because the CLI is interactive, the container cannot log to STDOUT as is customary for Docker containers. The KSQL CLI logs can instead be found at /var/logs/ksql-cli/ inside the container.
+
+Configuring the KSQL CLI
+""""""""""""""""""""""""
+* Config files: Assuming that the CLI's configuration file is available on the Docker host at ``/docker/container/path/ksql-cli.properties``, then the following command will make the configuration file available to the CLI container at ``/docker/container/path/ksql-cli.properties``.
+
+.. sourcecode:: bash
+  docker run -it -rm \
+    -v /docker/host/path/:/docker/container/path
+    confluentinc/cp-ksql-cli:4.1.2 \
+      http://ksql-server.host:8088 \
+      --config-file /docker/container/path/ksql-cli.properties
+
+
+* Additional CLI options can be appended to the command as necessary. See :ref:`the KSQL CLI documentation <install_ksql-cli>` for the KSQL CLI options you can use.
+* The KSQL CLI cannot be configured via environment variables, unlike the KSQL Server image.
```

Comment:
Why don't we allow configuring the CLI image via env variables?

Reply:
It's based on my understanding of the way docker operates. It allows only one command to execute at the end. For other images, that is a script, which reads the environment, creates config files, and then runs the actual command.

In this case, the command is our `ksql` cli command. That doesn't leave any space for setting up config files. 

I may, however, be wrong about this limitation. I couldn't get it to work, but that doesn't mean it can't. We can fix it in a future version of the image, though.